packages/react-text/src/components/Text/useText.ts

```diff
@@ -12,17 +10,29 @@ const mergeProps = makeMergeProps<TextState>();
  *
  * @param props - props from this instance of Text
  * @param ref - reference to root HTMLElement of Text
- * @param defaultProps - (optional) default prop values provided by the implementing type
  */
-export const useText = (props: TextProps, ref: React.Ref<HTMLElement>, defaultProps?: TextProps): TextState => {
-  const state = mergeProps(
-    {
-      ref,
-      as: 'span',
-    },
-    defaultProps,
-    props,
-  );
+export const useText = (props: TextProps, ref: React.Ref<HTMLElement>): TextState => {
+  const asProp = props.as || 'span';
+
+  return {
+    wrap: true,
+    truncate: false,
+    block: false,
+    italic: false,
+    underline: false,
+    strikethrough: false,
+    size: 300,
+    font: 'base',
+    weight: 'regular',
+    align: 'start',
 
-  return state;
+    components: { root: asProp },
```

Comment:
```suggestion
    components: { root: 'span' },
```

This configures slot's default behavior, should be hardcoded

> state.components should have hardcoded values with matching element
#18844

Reply:
How do we allow the user to override the default rendered element?

Reply:
In this case:
- via `as` prop, it should be handled via `getSlots()`, i.e. `<Text as="h1" />` still should work
- via composing new component:

```tsx
function TextAsH1() {
  const state = useText()
  state.components.root = 'h1'
} 
```

Is there anything that breaks this scenario?

Reply:
From what I'm seeing `getSlot` is picking the `as` prop from the slot in the state to get the element but `getNativeElementProps` strips `as` out as it's not a native prop. Should I not be using `getNativeElementProps`?

Reply:
Then let's keep it as it is for now. Can you please create an issue to track it? 🙏 It's not expected behavior

Reply:
Issue created https://github.com/microsoft/fluentui/issues/19785